finished/test.c

```diff
@@ -11,34 +12,49 @@ void runTests(){
   int pass_count = 0;
 
   stack *s = stack_new();
-  int *x = malloc(sizeof(int));
-  *x = 111;
-  int *y = malloc(sizeof(int));
-  *y = 222;
+  int x = 111;
+  int y = 222;
 
   TEST(1, (int*) stack_top(s) == NULL);
 
-  stack_push(s, x);
-  TEST(2, (int*) stack_top(s) == x);
+  stack_push(s, &x);
+  TEST(2, (int*) stack_top(s) == &x);
 
-  stack_push(s, y);
-  TEST(3, (int*) stack_top(s) == y);
+  stack_push(s, &y);
+  TEST(3, (int*) stack_top(s) == &y);
 
   TEST(4, stack_size(s) == 2);
 
   int *result = (int*) stack_pop(s);
-  TEST(5, result == y);
+  TEST(5, result == &y);
   TEST(6, *result == 222);
   TEST(7, stack_size(s) == 1);
 
-  result = (int*) stack_pop(s);
-  TEST(8, result == x);
+  result = (int*)stack_pop(s);
+  TEST(8, result == &x);
   TEST(9, *result == 111);
   TEST(10, stack_size(s) == 0);
 
+  for (int i = 0; i <= 10; i++) {
+    int *x = calloc(1, sizeof(int));
+    *x = 10 - i;
+    stack_push(s, x);
+  }
+  {
+    iter_t *it = iterator_new(s);
+    int cnt = 0;
+    while (iterator_more(it)) {
+      int *x = iterator_next(it);
+      TEST(11, *x == cnt);
+      free(x);
+      cnt++;
+    }
+    iterator_free(it);
+  }
   printf("Passed %d out of %d tests\n", pass_count, test_count);
+
+  stack_free(s);
+  assert(pass_count == test_count);
```

Comment:
> You can distinguish between tests crashing and running to completion (with zero or more failing).

Seems like a theoretical requirement. Why would I want to distinguish between my program being broken and... my program being broken? I'd rather distinguish between broken programs and non-broken ones.

> If you have several test files that you run via make, you will keep running them even if some tests in one file fail.

Which *I* don't want, in the default case. If I really want that: sure, I'll run `./test || true` instead. This has the semantics of before. The point is: now there's a choice. You can ignore the return value or not, but the value is meaningful.